openstudiocore/src/energyplus/ReverseTranslator/ReverseTranslateEnergyManagementSystemActuator.cpp

```diff
@@ -0,0 +1,100 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2016, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#include "../ReverseTranslator.hpp"
+
+#include "../../model/EnergyManagementSystemActuator.hpp"
+#include "../../model/EnergyManagementSystemActuator_Impl.hpp"
+
+#include <utilities/idd/EnergyManagementSystem_Actuator_FieldEnums.hxx>
+#include "../../utilities/idd/IddEnums.hpp"
+#include <utilities/idd/IddEnums.hxx>
+
+using namespace openstudio::model;
+
+namespace openstudio {
+
+namespace energyplus {
+
+OptionalModelObject ReverseTranslator::translateEnergyManagementSystemActuator(const WorkspaceObject & workspaceObject)
+{
+  if (workspaceObject.iddObject().type() != IddObjectType::EnergyManagementSystem_Actuator) {
+    LOG(Error, "WorkspaceObject is not IddObjectType: EnergyManagementSystem_Actuator");
+    return boost::none;
+  }
+
+  OptionalString s1 = workspaceObject.getString(EnergyManagementSystem_ActuatorFields::Name);
+  if(!s1){
+    LOG(Error, "WorkspaceObject EnergyManagementSystem_Actuator has no name");
+    return boost::none;
+  }
+
+  OptionalString s = workspaceObject.getString(EnergyManagementSystem_ActuatorFields::ActuatedComponentUniqueName);
+  if (!s) {
+    LOG(Error, workspaceObject.nameString() + ": has no ActuatedComponentUniqueName");
+    return boost::none;
+  }
+
+  Workspace workspace = workspaceObject.workspace();
+
+  if (s) {
+    for ( WorkspaceObject& wsObject : workspace.getObjectsByName(*s)) {
```

Comment:
Is this really what you want to do here in the case there getObjectsByName(*s) returns multiple objects?  Maybe trap the case that it returns > 1 object and return an error?